### PR TITLE
Added FCM support for Android

### PR DESCRIPTION
**Pull request by v-fekalu**

### CODE REVIEW

File: Android/notification-hubs-sdk-e2etestapp/notification-hubs-sdk-e2etestapp.iml [hunk 1]
```diff
@@ -57,51 +57,40 @@
...
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-annotations:25.3.1@jar" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:animated-vector-drawable-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-compat-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-v4-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-vector-drawable-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-core-ui-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-media-compat-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-core-utils-25.3.1" level="project" />
+    <orderEntry type="library" exported="" name="Gradle: com.android.support:appcompat-v7-21.0.3" level="project" />
```
Comment (stankovski) [inline Android/notification-hubs-sdk-e2etestapp/notification-hubs-sdk-e2etestapp.iml]: Do we need this change?

File: Android/notification-hubs-sdk-tests/notification-hubs-sdk-tests.iml [hunk 2]
```diff
@@ -86,16 +86,9 @@
...
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-annotations:25.3.1@jar" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:animated-vector-drawable-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-compat-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-v4-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-vector-drawable-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-core-ui-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-media-compat-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-core-utils-25.3.1" level="project" />
+    <orderEntry type="library" exported="" name="Gradle: com.android.support:appcompat-v7-21.0.3" level="project" />
```
Comment (stankovski) [inline Android/notification-hubs-sdk-tests/notification-hubs-sdk-tests.iml]: Do we need this change?

File: Android/notification-hubs-sdk/notification-hubs-sdk.iml [hunk 3]
```diff
@@ -58,56 +58,50 @@
...
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-annotations:25.3.1@jar" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:animated-vector-drawable-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-compat-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-v4-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-vector-drawable-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-core-ui-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-media-compat-25.3.1" level="project" />
-    <orderEntry type="library" exported="" name="Gradle: com.android.support:support-core-utils-25.3.1" level="project" />
+    <orderEntry type="library" exported="" name="Gradle: com.android.support:appcompat-v7-21.0.3" level="project" />
```
Comment (stankovski) [inline Android/notification-hubs-sdk/notification-hubs-sdk.iml]: Same question

File: Android/notification-hubs-sdk/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -4,9 +4,9 @@
     android:versionCode="1"
     android:versionName="1.0" >
 
-    <uses-sdk
+<!--    <uses-sdk
```
Comment (stankovski) [inline Android/notification-hubs-sdk/src/main/AndroidManifest.xml]: Why this change?

File: Android/notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/FcmTemplateRegistration.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+package com.microsoft.windowsazure.messaging;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+/**
+ * Represents FCM template registration
+ */
+public class FcmTemplateRegistration extends TemplateRegistration {
```
Comment (stankovski) [inline Android/notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/FcmTemplateRegistration.java]: Need to mark `FcmTemplateRegistration` and `FcmNativeRegistration` classed as deprecated

File: Android/notification-hubs-sdk/src/main/AndroidManifest.xml (hunk 4 above)
Comment (v-fekalu) [reply Android/notification-hubs-sdk/src/main/AndroidManifest.xml]: Not able to compile with latest Android studio

File: Android/notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/FcmTemplateRegistration.java (hunk 5 above)
Comment (v-fekalu) [reply Android/notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/FcmTemplateRegistration.java]: Ok, but maybe GcmTemplateRegistration ?
